### PR TITLE
Rename hcloud_*_facts to hcloud_*_info

### LINKED ISSUE #60516: Hetzner Cloud _facts modules need to be renamed to _info and stop returning ansible_facts
##### SUMMARY
According to the [requirements](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements), `_facts` modules should return information that is specific to the host the module runs on as `ansible_facts`. (See also ansible/ansible#54280 for some discussion.)

I think that the following modules do not satisfy this condition: they return information that is not specific to the host machine, but bound to a cloud account. Please verify whether this is actually the case.

To fix this, the core team thinks the best approach is to rename the `_facts` modules to `_info` modules, and change the return value so that if the module is called with the `_facts` name (should be deprecated, and will then be removed in Ansible 2.13), it should return `ansible_facts` as now (and print a deprecation message), and if it is called with the `_info` name, it no longer returns `ansible_facts`, so that the result needs to be registered. This informs users that they have to change something, won't break their playbooks, and allows them (by switching to the `_info` names) to verify that their roles and playbooks no longer use the facts.

If that approach doesn't work, or you want to change the module's interface (like rename the return value(s)), it is better to deprecate the `_facts` module and create a new `_info` module next to it. To see both approaches implemented, you can take a look at https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f (ansible/ansible#60178).

If you have any input — preferred method to solve this, doubts that all modules should be renamed, ... — please write it here!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
hcloud_datacenter_facts
hcloud_floating_ip_facts
hcloud_image_facts
hcloud_location_facts
hcloud_server_facts
hcloud_server_type_facts
hcloud_ssh_key_facts
hcloud_volume_facts

##### ANSIBLE VERSION
```paste below
2.9.0
```

### PR DESCRIPTION
**Pull request by LKaemmerling**
##### SUMMARY

This will rename all `hcloud_*_facts` modules to `*_info`.

Closes #60516

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- [X]  hcloud_datacenter_facts
- [x]  hcloud_floating_ip_facts
- [X]  hcloud_image_facts
- [X]  hcloud_location_facts
- [x]  hcloud_server_facts
- [x]  hcloud_server_type_facts
- [x]  hcloud_ssh_key_facts
- [x]  hcloud_volume_facts

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/hcloud/hcloud_datacenter_info.py [hunk 1]
```diff
@@ -93,13 +95,13 @@
 
 class AnsibleHcloudDatacenterFacts(Hcloud):
```
Comment (LKaemmerling) [inline lib/ansible/modules/cloud/hcloud/hcloud_datacenter_info.py]: Should be renamed too

**Review by ghost — APPROVED** [review]
LGTM

File: lib/ansible/modules/cloud/hcloud/hcloud_datacenter_info.py [hunk 2]
```diff
@@ -40,32 +42,35 @@
...
+    var: output
 """
 
 RETURN = """
-hcloud_datacenter_facts:
-    description: The datacenter facts as list
+hcloud_datacenter_info:
+    description:
+      - The datacenter info as list
+      - This module was called C(hcloud_datacenter_facts) before Ansible 2.9, returning C(ansible_facts) and C(hcloud_datacenter_facts).
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/hcloud/hcloud_datacenter_info.py:57]:
```suggestion
      - This module was called C(hcloud_datacenter_facts) before Ansible 2.9, returning C(hcloud_datacenter_facts) as C(ansible_facts).
```
You have to decide whether you want to change this in all places (or in none :) ).

**Review by felixfontein — COMMENTED** [review]
LGTM. I've found one minor detail; if you want to change it, please do so.

What's missing are porting guide entries and a changelog. See 2d98734ad51c3de557c3a7d1f7166bdd114ceb0f for how this should look like.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@LKaemmerling @cschmitt-hcloud thanks for working on this!

**Pull request merged**